File: src/main/java/com/lanjii/biz/admin/system/model/entity/SysRole.java
package com.lanjii.biz.admin.system.model.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import com.lanjii.biz.admin.system.model.dto.SysRoleDTO;
import com.lanjii.biz.admin.system.model.vo.SysRoleVO;
import com.lanjii.core.base.BaseEntity;
import com.lanjii.core.base.BaseEntityMapper;
import lombok.Data;
import lombok.EqualsAndHashCode;
import org.mapstruct.Mapper;
import org.mapstruct.Mapping;
import org.mapstruct.factory.Mappers;

/**
 * 角色表(SysRole)表实体类
 *
 * @author lanjii
 */
@Data
@EqualsAndHashCode(callSuper = true)
@TableName("sys_role")
public class SysRole extends BaseEntity<SysRole> {

    /**
     * 角色ID
     */
    private Long id;

    /**
     * 角色名称
     */
    private String name;

    /**
     * 角色编码
     */
    private String code;

    /**
     * 显示顺序
     */
    private Integer sortOrder;

    /**
     * 是否启用（1启用 0禁用）
     */
    private Integer isEnabled;

    /**
     * 备注
     */
    private String remark;


    @Mapper
    public interface SysRoleMapper extends BaseEntityMapper<SysRole, SysRoleVO, SysRoleDTO> {

        @Mapping(target = "isEnabledLabel", expression = "java(dictValueToLabel(entity.getIsEnabled(),\"STATUS\"))")
        SysRoleVO toVo(SysRole entity);

    }

    public static final SysRoleMapper INSTANCE = Mappers.getMapper(SysRoleMapper.class);


}

File: ui/lanjii-ui/src/types/sys/sysRole.ts
/**
 * 系统角色实体（对应后端SysRole）
 */
export interface SysRole {
  /** 角色ID */
  id?: number
  /** 角色名称 */
  name: string
  /** 角色编码 */
  code: string
  /** 显示顺序 */
  sortOrder?: number
  /** 是否启用（1启用 0禁用） */
  isEnabled?: number
  /** 备注 */
  remark?: string
  /** 启用状态标签（字典转换） */
  isEnabledLabel?: string
  /** 创建时间 */
  createTime?: string
  /** 更新时间 */
  updateTime?: string
  /** 创建人 */
  createBy?: string
  /** 更新人 */
  updateBy?: string
  /** 删除标志（0-未删除，1-已删除） */
  deleted?: number
}
